### PR TITLE
Feature/trieuth/t03 add pagination

### PR DESCRIPTION
**Pull request by HoangTrieu2803**

### CODE REVIEW

File: src/containers/Home/Jobs/ListJob/index.tsx [hunk 1]
```diff
@@ -1,33 +1,29 @@
...
+                    <li key={key} className={`list-group-item d-flex job`} data-testid="test100Jobs">
                         <img className='job__img' alt='logo' src={item.img} />
                         <div className='job__tittle'>
                             <h4>{item.jobName}</h4>
-                            <ul className='job__tech'>
-                                <li>Html</li>
-                                <li>Css</li>
-                                <a href="/#">Sign in to view</a>
-                            </ul>
+                            <p className='job__tech'>
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/ListJob/index.tsx:15]: wrap with `div` element

File: src/containers/Home/Jobs/ListJob/index.tsx [hunk 2]
```diff
@@ -1,33 +1,29 @@
...
+                                
+                                <a href="/#"><FontAwesomeIcon className='job__icon' icon={faDollarSign}/> Sign in to view <FontAwesomeIcon icon={faCaretRight}/></a>
+                                <button>Css</button>
+                                <button>Html</button>
+                                <button>Javascript</button>
+                                <button>Fresher accept</button>
+                            </p>
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/ListJob/index.tsx:22]: wrap with `div` don't use `p`

File: src/containers/Home/Jobs/Pagination/index.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+export default function Pagination(props: { jobsPerPage: number, totalJobs: number, paginate: any, currentPage:number }) {
+    const { jobsPerPage, totalJobs, paginate,currentPage } = props
+    const pageNumber = []
+    for (let i = 1; i <= Math.ceil(totalJobs / jobsPerPage); i++) {
+        pageNumber.push(i)
+    }
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/Pagination/index.tsx:6]: should use useEffect to handle this logic

File: src/containers/Home/Jobs/index.tsx [hunk 4]
```diff
@@ -1,15 +1,56 @@
...
+                    <ul className='container-fluid mb-2' id='listJobs'>
+                        <ListJob currentJobs={currentJobs} />
+                    </ul>
+                    <nav>
+                        <ul className='pagination'>
+                            <button className="btn btn-danger" onClick={() => {
+                                setCurrentPage(currentPage - 1)
+                                if(currentPage == 1){
+                                    setCurrentPage(10)
+                                }
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/index.tsx:41]: change the logic here

File: src/containers/Home/Jobs/index.tsx [hunk 5]
```diff
@@ -1,15 +1,56 @@
...
-            </nav>
+            <h2 data-testid="jobTittle">{jobs.length} Jobs For "Chất" Developers</h2>
+            {loading ? <Loading /> :
+                <>
+                    <ul className='container-fluid mb-2' id='listJobs'>
+                        <ListJob currentJobs={currentJobs} />
+                    </ul>
+                    <nav>
+                        <ul className='pagination'>
+                            <button className="btn btn-danger" onClick={() => {
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/index.tsx:37]: should create a function like: `handleClickPreviousPage` to easy when reading the code

File: src/containers/Home/Jobs/index.tsx [hunk 6]
```diff
@@ -1,15 +1,56 @@
...
+                                setCurrentPage(currentPage - 1)
+                                if(currentPage == 1){
+                                    setCurrentPage(10)
+                                }
+                            }}><FontAwesomeIcon icon={faArrowLeft} /></button>
+                            <Pagination jobsPerPage={jobsPerPage} totalJobs={jobs.length} paginate={paginate} currentPage={currentPage} />
+                            <button className="btn btn-danger" onClick={() => {
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/index.tsx:44]: the same comment above

File: src/containers/Home/Jobs/ListJob/index.tsx [hunk 7]
```diff
@@ -1,33 +1,29 @@
...
-    const { data: jobs, loading } = useSelector((state: RootState) => state.jobs)
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faDollarSign, faCaretRight } from '@fortawesome/free-solid-svg-icons';
+export default function ListJob(props:{currentJobs:Job[]}) {
+    const {currentJobs} = props
     return (
         <div>
-            <h2 data-testid="test100Job">{jobs.length} Jobs For "Chất" Developers</h2>
-            {loading ? <Loading /> : jobs?.map((item: Job, key: number) => {
+            {currentJobs?.map((item: Job, key: number) => {
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/ListJob/index.tsx]: i've changed

File: src/containers/Home/Jobs/ListJob/index.tsx (hunk 2 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/ListJob/index.tsx]: i've changed

File: src/containers/Home/Jobs/Pagination/index.tsx (hunk 3 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/Pagination/index.tsx]: i've changed

File: src/containers/Home/Jobs/index.tsx [hunk 8]
```diff
@@ -1,15 +1,56 @@
...
-            <nav >
-                <ul className='pagination'>
-                </ul>
-            </nav>
+            <h2 data-testid="jobTittle">{jobs.length} Jobs For "Chất" Developers</h2>
+            {loading ? <Loading /> :
+                <>
+                    <ul className='container-fluid mb-2' id='listJobs'>
+                        <ListJob currentJobs={currentJobs} />
+                    </ul>
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i've moved

File: src/containers/Home/Jobs/index.tsx (hunk 5 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i've created function for this

File: src/containers/Home/Jobs/Pagination/index.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    },[totalPage])
+
+    return (
+        <>
+            {pageNumber.map((item,index) => {
+                return (
+                    <li className='page-item pagination__item' data-testid="pagination" key={index} >
+                        <a onClick={() => {
+                            paginate(item)                          
+                        }} href="/#" className={`page-link pagination__item__link ${item==currentPage ? "pagination__item__active" : ""}`}>
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/Pagination/index.tsx:23]: should `pagination__item--active`

File: src/containers/Home/Jobs/index.tsx [hunk 10]
```diff
@@ -1,15 +1,52 @@
...
-                <ul className='pagination'>
-                </ul>
-            </nav>
+            <h2 data-testid="jobTittle">{jobs.length} Jobs For "Chất" Developers</h2>
+            {loading ? <Loading /> :
+                <>
+                    <ListJobs currentJobs={currentJobs} />
+                    <nav>
+                        <ul className='pagination'>
+                            <button className="btn btn-danger" disabled={currentPage == 1 ? true : false} onClick={handleClickPreviousPage}><FontAwesomeIcon icon={faArrowLeft} /></button>
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/index.tsx:43]: use strict comparison

File: src/containers/Home/Jobs/index.tsx [hunk 11]
```diff
@@ -1,15 +1,52 @@
...
+                            <Pagination jobsPerPage={jobsPerPage} totalJobs={jobs.length} paginate={paginate} currentPage={currentPage} />
+                            <button className="btn btn-danger" disabled={currentPage == 10 ? true : false} onClick={handleClickNextPage}><FontAwesomeIcon icon={faArrowRight} /></button>
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/index.tsx:45]: the same above

File: src/containers/Home/Jobs/Pagination/index.tsx (hunk 9 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/Pagination/index.tsx]: i've changed name

File: src/containers/Home/Jobs/index.tsx (hunk 11 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i've done

**Pull request merged**